csharp/HOLMS.Platform/HOLMS.Platform/Support/Currency/DollarCents.cs

```diff
@@ -0,0 +1,94 @@
+
+            IsNegative = dc.IsNegative;
+            Dollars = dc.Dollars;
+            Cents = dc.Cents;
+        }
+
+        public static DollarCents FromCents(int cents) {
+            var isneg = cents < 0;
+
+            var posCents = (isneg ? -1 : 1) * cents;
+            var wholeDollars = (uint)posCents / 100;
+            var centsLeft = (uint)(posCents - wholeDollars * 100);
+
+            return new DollarCents(isneg, wholeDollars, centsLeft);
+        }
+
+        public static DollarCents Zero => new DollarCents(false, 0, 0);
+
+        public int TotalCents => (int)((IsNegative ? -1 : 1) * (Dollars * 100 + Cents));
+        public decimal AsDecimal => TotalCents / 100m;
+
+        public MonetaryAmount ToPb => new MonetaryAmount {
+            Microdollars = TotalCents * OneMillion
+        };
+
+        private static int CompareTo(DollarCents x, DollarCents y) {
+            var isNegativeComparison = x.IsNegative.CompareTo(y.IsNegative);
+            if (isNegativeComparison != 0) {
+                return isNegativeComparison;
+            }
+
+            var dollarsComparison = x.Dollars.CompareTo(y.Dollars);
+            if (dollarsComparison != 0) {
+                return dollarsComparison;
+            }
+
+            return x.Cents.CompareTo(y.Cents);
+        }
+
+        // h/t https://stackoverflow.com/questions/20593755/why-do-i-have-to-overload-operators-when-implementing-compareto
+        public int CompareTo(DollarCents x) { return CompareTo(this, x); }
+        public bool Equals(DollarCents x) { return CompareTo(this, x) == 0; }
+        public static bool operator <(DollarCents x, DollarCents y) { return CompareTo(x, y) < 0; }
+        public static bool operator >(DollarCents x, DollarCents y) { return CompareTo(x, y) > 0; }
+        public static bool operator <=(DollarCents x, DollarCents y) { return CompareTo(x, y) <= 0; }
+        public static bool operator >=(DollarCents x, DollarCents y) { return CompareTo(x, y) >= 0; }
+        public static bool operator ==(DollarCents x, DollarCents y) { return CompareTo(x, y) == 0; }
+        public static bool operator !=(DollarCents x, DollarCents y) { return CompareTo(x, y) != 0; }
+
+        public static DollarCents operator +(DollarCents a, DollarCents b) => FromCents(a.TotalCents + b.TotalCents);
+        public static DollarCents operator -(DollarCents a, DollarCents b) => FromCents(a.TotalCents - b.TotalCents);
+
+        public override bool Equals(object obj) {
+            return (obj is DollarCents) && (CompareTo(this, (DollarCents)obj) == 0);
+        }
+
+        public override int GetHashCode() {
+            unchecked {
+                var hashCode = IsNegative.GetHashCode();
+                hashCode = (hashCode * 397) ^ (int)Dollars;
+                hashCode = (hashCode * 397) ^ (int)Cents;
+                return hashCode;
+            }
+        }
+
+        public override string ToString() {
```

Comment:
.Net has an interface `IFormattable` that would be the "cross your i's dot your t's" way of implementing this. It describes how the object responds to a format string. The doc on this is [here](https://msdn.microsoft.com/en-us/library/system.iformattable.tostring(v=vs.110).aspx).

I don't think it's 100% necessary, especially an exhaustive implementation, but it would be nice feature for consumers of this class to be able to specify "C0" vs "C2". It would also make life in WPF much easier, because the string formatting (C2 vs C0) is specified in the xaml, and you could strip decimals completely out of the databinding, and instead databind directly onto this datatype, and the string formatting would work identically.